src/modules/billing/services/billing-volumes-service.js

```diff
@@ -152,12 +152,16 @@ const getBillingVolumesByChargeVersion = async (chargeVersionId, financialYear,
  * generated
  *
  * @param {BillingVolume} billingVolume
- * @return {Promise<DateRange>}
+ * @return {Promise<DateRange>, invoiceAccountId} invoiceAccountId is optional for the ui
  */
-const getBillingVolumeChargePeriod = async billingVolume => {
+const getBillingVolumeChargePeriod = async (billingVolume, includeTailData = false) => {
   const chargeVersionId = get(billingVolume, 'chargeElement.chargeVersionId');
   const chargeVersion = await chargeVersionService.getByChargeVersionId(chargeVersionId);
-  return chargePeriod.getChargePeriod(billingVolume.financialYear, chargeVersion);
+  if (includeTailData) {
```

Comment:
Since getting the `invoiceAccountId` doesn't involve another DB query, I think it might be simpler to remove this conditional logic and get the consumers of the function to extract what they need from the return value.

I think this would also increase the test coverage slightly.